File: ruby/lib/ccpayment/services/user_transfer_service.rb
module CCPayment
  module Services
    class UserTransferService
      def initialize(client)
        @client = client
      end

      def user_transfer(data)
        @client.post('/userTransfer', data)
      end

      def get_user_transfer_record(data)
        @client.post('/getUserTransferRecord', data)
      end

      def get_user_transfer_record_list(data = {})
        @client.post('/getUserTransferRecordList', data)
      end

      def user_batch_transfer(data)
        @client.post('/userBatchTransfer', data)
      end

      def get_user_batch_transfer_record(data)
        @client.post('/getUserBatchTransferRecord', data)
      end

      def get_user_batch_transfer_record_list(data = {})
        @client.post('/getUserBatchTransferRecordList', data)
      end

      def get_user_batch_transfer_record_detail(data)
        @client.post('/getUserBatchTransferRecordDetail', data)
      end
    end
  end
end

File: php/src/Services/UserTransferService.php
<?php

namespace CCPayment\Services;

use CCPayment\Client;
use CCPayment\APIError;
use GuzzleHttp\Exception\GuzzleException;

class UserTransferService
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function userTransfer(array $data): array
    {
        return $this->client->post('/userTransfer', $data);
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function getUserTransferRecord(array $data): array
    {
        return $this->client->post('/getUserTransferRecord', $data);
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function getUserTransferRecordList(array $data = []): array
    {
        return $this->client->post('/getUserTransferRecordList', $data);
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function userBatchTransfer(array $data): array
    {
        return $this->client->post('/userBatchTransfer', $data);
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function getUserBatchTransferRecord(array $data): array
    {
        return $this->client->post('/getUserBatchTransferRecord', $data);
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function getUserBatchTransferRecordList(array $data = []): array
    {
        return $this->client->post('/getUserBatchTransferRecordList', $data);
    }

    /**
     * @throws APIError
     * @throws GuzzleException
     */
    public function getUserBatchTransferRecordDetail(array $data): array
    {
        return $this->client->post('/getUserBatchTransferRecordDetail', $data);
    }
}
